### PR TITLE
Recommendations for new dataset page

### PR DESCRIPTION
**Pull request by nicolaskempf57**
See https://github.com/etalab/data.gouv.fr/issues/1000

udata-recommendations is using `dataset/card.html` but this file won't exist after https://github.com/etalab/udata-front/pull/181.

### CODE REVIEW

File: udata_recommendations/templates/dataset-recommendations.html [hunk 1]
```diff
@@ -19,15 +26,19 @@ <h3>Réutilisations recommandées</h3>
 {% endif -%}
 {%- if reco_datasets %}
 <div id="dataset-recommendations" class="fr-hidden fr-unhidden-sm">
-    <h3>Jeux de données recommandés</h3>
+    <h3 class="subtitle subtitle--uppercase fr-m-0">
+        {{ ngettext('%(num)d recommended dataset', '%(num)d recommended datasets', reco_datasets|length) }}
```
Comment (maudetes) [inline udata_recommendations/templates/dataset-recommendations.html:30]: You'll need to extract translations! ;)

File: tests/test_views.py [hunk 2]
```diff
@@ -74,7 +74,7 @@ def test_view_dataset_with_recommendations_dedupe(self, datasets, reuses):
 
         content = render_hook(dataset=dataset)
 
-        assert content.count(f'href="{ds1.external_url}"') == 1
+        assert content.count(f'href="{ds1.display_url}"') == 1
```
Comment (maudetes) [inline tests/test_views.py:77]: I did not see where this modification comes from

**Review by maudetes — COMMENTED** [review]
We should wait for https://github.com/etalab/udata-front/pull/181 to be merged before merging this one?

**Comment by nicolaskempf57** [comment]
> We should wait for [etalab/udata-front#181](https://github.com/etalab/udata-front/pull/181) to be merged before merging this one?

Yes ! This will allow our tests to pass too.

File: tests/test_views.py (hunk 2 above)
Comment (nicolaskempf57) [reply tests/test_views.py:77]: This is coming from [udata-front card-sm](https://github.com/etalab/udata-front/pull/181/files#diff-509bf4b5ff62f9953a8998252980fd626b73042c4dcdb984bf1c014607739695R52).

Previously, we had a card template with an external URL and a search-result template without.

Now, both are using the same link (`url_for('datasets.show', dataset=dataset)`).

File: .gitignore [hunk 3]
```diff
@@ -49,7 +49,6 @@ coverage.xml
 
 # Translations
 *.mo
-*.pot
```
Comment (nicolaskempf57) [inline .gitignore:52]: This one is added to git to match others repos (udata and udata-front)

File: udata_recommendations/templates/dataset-recommendations.html (hunk 1 above)
Comment (nicolaskempf57) [reply udata_recommendations/templates/dataset-recommendations.html:30]: done !

File: requirements/install.pip [hunk 4]
```diff
@@ -1,3 +1,3 @@
 udata>=3.2.0
-udata-front>=1.0.0
+git+https://github.com/nicolaskempf57/udata-front@feat/new-dataset-page#egg=udata_front
```
Comment (maudetes) [inline requirements/install.pip:2]: Let's not forget ;)

File: udata_recommendations/tasks.py [hunk 5]
```diff
@@ -106,11 +108,23 @@ def process_dataset(source, dataset):
             except (Reuse.DoesNotExist, mongoengine.errors.ValidationError):
                 error(f"Recommended reuse {reco['id']} not found")
                 continue
+        elif reco_type == 'external':
+            try:
+                external = urlsplit(reco['id']).geturl()
```
Comment (maudetes) [inline udata_recommendations/tasks.py:113]: Do you use it to validate the url? You could use the helper `from udata.uris import validate` if we want to be consistent on our URL validation?

File: udata_recommendations/views.py [hunk 6]
```diff
@@ -10,28 +11,66 @@
 
 
 def has_recommendations(ctx):
```
Comment (maudetes) [inline udata_recommendations/views.py:13]: Should we make it explicit, ex: `has_dataset_recommendations`?

**Review by maudetes — APPROVED** [review]
It seems to work nicely!

Don't forget the changelog

File: udata_recommendations/tasks.py (hunk 5 above)
Comment (nicolaskempf57) [reply udata_recommendations/tasks.py]: Thx !

**Pull request merged**